agent/agent.go

```diff
@@ -18,6 +18,12 @@ const (
 	nodeUpdatePeriod             = 20 * time.Second
 )
 
+// Secrets is a map that keeps all the currenty available secrets to the agent
+type Secrets struct {
```

Comment:
That contains just this secret struct with its APIs, or also all the worker apis that also deal with secrets?